kyaml/kio/byteio_reader.go

```diff
@@ -164,15 +164,13 @@ type ByteReader struct {
 
 var _ Reader = &ByteReader{}
 
-// splitDocuments returns a slice of all documents contained in a YAML string. Multiple documents can be divided by the
+// SplitDocuments returns a slice of all documents contained in a YAML string. Multiple documents can be divided by the
 // YAML document separator (---). It allows for white space and comments to be after the separator on the same line,
 // but will return an error if anything else is on the line.
-func splitDocuments(s string) ([]string, error) {
+func SplitDocuments(s string) ([]string, error) {
```

Comment:
Since we're exposing this, can you please add some basic tests for it?

Reply:
Tests added